### PR TITLE
add OpenShift README How To/YAML & .s2i/environment

### PR DESCRIPTION
**Pull request by vorburger**
see https://github.com/alfio-event/alf.io/issues/403

### CODE REVIEW

**Comment by vorburger** [comment]
@cbellone @syjer perhaps you would like to (manually) test this? Just shout if NOK for you...

**Comment by vorburger** [comment]
@coveralls how can the coverage decrease if this adds no new code?! :smile_cat: 

@cbellone @syjer perhaps you would like to make Coverall less picky and shut up for <1% ?

**Comment by vorburger** [comment]
@syjer @cbellone so how do you like this? Anything else I can do to get this PR merged?

File: README.md [hunk 1]
```diff
@@ -72,9 +72,36 @@ Importing the Gradle project into Intellij and Eclipse both work.
 
 `./gradlew dependencyUpdates`
 
+
+## Run on OpenShift
```
Comment (cbellone) [inline README.md]: I would prefer to have this section in another tutorial or wiki page, as we don't have direct reference to other providers on the README.
However, we could add a section "How to run in production", linking the tutorials on the website. WDYT?

File: openshift.yaml [hunk 2]
```diff
@@ -0,0 +1,196 @@
+apiVersion: v1
```
Comment (cbellone) [inline openshift.yaml]: is it possible to move this in a subdirectory, like etc/openshift/ ?

**Comment by vorburger** [comment]
@cbellone yes! On my list, I'll see if I can make time for this afternoon...

**Comment by vorburger** [comment]
@cbellone I should have a moment today and tomorrow (but not next week) for this - shall we try to wrap it up and get it in? I'll rebase it and address your comments... just noticed that I did this for `1.x-maintenance`.. do you still maintain that, or should I now make this for `master` instead?

File: README.md (hunk 1 above)
Comment (vorburger) [reply README.md]: sure, I can move it out of the main README and just link to it.  But I'm personally [not a huge fan of keeping doc in a separate repo](https://github.com/alfio-event/alf.io) .. IMHO it's easier to have doc as an integral part of a PR together with code changes. I'm therefore going to propose a new `docs/` folder here - hope you'll like that.

File: openshift.yaml (hunk 2 above)
Comment (vorburger) [reply openshift.yaml]: yes this file can be anywhere you like; I'll move it to `etc/openshift/` as suggested.

**Comment by vorburger** [comment]
> do you still maintain that, or should I now make this for `master` instead?

actually that would take a little bit of work to adjust - how about first getting this finalized and into `1.x-maintenance`, and then I'll (try to) follow up with a cherry-pick to master and the necessary adjustments?

**Comment by vorburger** [comment]
> do you still maintain that, or should I now make this for `master` instead?

actually that would take a little bit of work to adjust - how about first getting this finalized and into `1.x-maintenance`, and then I'll (try to) follow up with a cherry-pick to master and the necessary adjustments?

**Comment by cbellone** [comment]
1.x-maintenance is the stable branch. We hope to release 2.0 by the end of this year, so we'll have plenty of time to update the tutorial :-)
RE: doc, I think it's OK, but then we'll have some documentation on the website and other content on the repo. I think that would be confusing for the users... WDYT?

**Comment by vorburger** [comment]
> RE: doc, I think it's OK, but then we'll have some documentation on the website and other content on the repo. I think that would be confusing for the users... WDYT?

I find the doc on the web site is kind of hard to find.. I don't suppose it would help to convince you if I volunteered to "port" (copy/paste) the Cloud Foundry and Heroku doc into `docs/deployment` as well? Otherwise if you feel strongly about this I guess I could also raise a PR against the `alf.io-website` instead - as per https://github.com/alfio-event/alf.io-website/issues/6. I just really don't like having to write HTML for this, see  :-( What would you like me to do so that we can wrap up this PR?

**Comment by cbellone** [comment]
I think we're OK here. And I also think that it doesn't make sense to duplicate content. 
We can start a separate discussion (maybe on the mailing list?) about how to organize the documentation. I would be great if you could be part of that.

Can you raise a PR against the alf.io-website linking the tutorial on the homepage and on the "tutorials" menu?

**Comment by vorburger** [comment]
> I think we're OK here. And I also think that it doesn't make sense to duplicate content.
> We can start a separate discussion (maybe on the mailing list?) about how to organize the documentation. I would be great if you could be part of that.

I would love to, but I probably won't have time, sorry... 

> Can you raise a PR against the alf.io-website linking the tutorial on the homepage and on the "tutorials" menu?

sure, see https://github.com/alfio-event/alf.io-website/pull/7

**Pull request merged**